File: solutions/go/grains/1/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(number int) (uint64, error) {
	if number <= 0 || number > 64 {
		return 0, errors.New("")
	}
	var result uint64 = 0
	result = uint64(math.Pow(2, float64(number-1)))
	return result, nil
}

func Total() uint64 {
	var result uint64 = 0
	for i := 1; i <= 64; i++ {
		var tmp, _ = Square(i)
		result += tmp
	}
	return result
}

File: solutions/lua/grains/1/grains.lua
local grains = {}

function grains.square(n)
    local result = 0.5
    while n > 0 do
        result = result * 2
        n = n - 1
    end
    return result
end

function grains.total()
    local result = 0
    for i = 1, 64, 1 do result = result + grains.square(i) end
    return result
end

return grains
